File: 3375-minimum-operations-to-make-array-values-equal-to-k/3375-minimum-operations-to-make-array-values-equal-to-k.cpp
#include <vector>
#include <set>

class Solution {
public:
    int minOperations(std::vector<int>& nums, int k) {
        for(auto num : nums){
            if(num < k){
                return -1;
            }
        }
        
        std::set<int> distinctGreater;
        for(auto num : nums){
            if(num > k){
                distinctGreater.insert(num);
            }
        }
        

        return distinctGreater.size();
    }
};

File: 3375-minimum-operations-to-make-array-values-equal-to-k/3375-minimum-operations-to-make-array-values-equal-to-k.java
class Solution {
    public int minOperations(int[] nums, int k) {
        for (int num : nums) {
            if (num < k) {
                return -1;
            }
        }

        Set<Integer> uniqueGreaterThanK = new HashSet<>();
        for (int num : nums) {
            if (num > k) {
                uniqueGreaterThanK.add(num);
            }
        }

        return uniqueGreaterThanK.size();
    }
}